community/pike/APKBUILD

```diff
@@ -0,0 +1,235 @@
+# Maintainer: Paweł Tomak <pawel@tomak.eu>
+pkgname=pike
+pkgver=8.0.610
+pkgrel=0
+pkgdesc="Pike Programing language"
+url="https://pike.lysator.liu.se"
+arch="x86_64"
+license="MIT"
+options="!check" # there is no tests for those packages
+depends=
+makedepends="gmp-dev \
+			bzip2-dev \
+			freeglut-dev \
+			gdbm-dev \
+			giflib-dev \
+			giflib-utils \
+			gtk+2.0 \
+			libgcrypt-dev \
+			libglade-dev \
+			libsm-dev \
+			mesa-gl \
+			nettle \
+			nettle-dev \
+			nettle-utils \
+			pcre-dev \
+			sdl-dev \
+			sdl_mixer-dev \
+			sqlite-dev \
+			sqlite-libs \
+			readline \
+			zlib-dev \
+			librsvg-dev"
+install=
+subpackages="$pkgname-bzip2 \
+			$pkgname-dnssd \
+			$pkgname-fuse \
+			$pkgname-gdbm \
+			$pkgname-gl \
+			$pkgname-gtk \
+			$pkgname-image \
+			$pkgname-pcre \
+			$pkgname-sdl \
+			$pkgname-svg \
+			$pkgname-sqlite \
+			$pkgname-dev \
+			$pkgname-doc \
+			$pkgname-core"
+source="http://pike.lysator.liu.se/pub/pike/all/${pkgver}/Pike-v${pkgver}.tar.gz"
+builddir="$srcdir/"
+
+unpack() {
+	cd "$builddir"
+	tar xzf Pike-v${pkgver}.tar.gz --strip-components=1
+}
+
+build() {
+	cd "$builddir"
+	make CONFIGUREARGS=" \
+		--prefix=/usr \
+		--disable-make_conf \
+		--disable-noopty-retry \
+		--without-cdebug \
+		--without-bundles \
+		--without-ssleay \
+		--with-gmp \
+		--with-crypt \
+		--with-bignums \
+		--without-rtldebug \
+		--with-Bz2 \
+		--without-fftw \
+		--without-_Ffmpeg \
+		--with-gdbm \
+		--without-GTK1 \
+		--with-GTK2 \
+		--with-jpeglib \
+		--without-Kerberos \
+		--without-msql \
+		--without-mysql \
+		--without-Odbc \
+		--without-oracle \
+		--with-GL \
+		--with-GLUT \
+		--with-_Regexp_PCRE \
+		--without-libpdf \
+		--without-sass \
+		--with-sane \
+		--with-SDL \
+		--with-SDL_Mixer \
+		--with-svg \
+		--with-tifflib \
+		--without-freetype \
+		--without-vcdiff \
+		--with-_Image_WebP \
+		--with-zlib"
+}
+
+package() {
+	cd "$builddir"
+	depends=""
+	for pkg in $(printf "%s" "$subpackages" | tr ' ' '\n'); do
+		depends="$depends $pkg=$pkgver-r$pkgrel"
+	done
+	make INSTALLARGS="--traditional" buildroot="$pkgdir" install_nodoc
+}
+
+_move_item() {
+	local bn=$(basename "$1")
+	local dn=$(dirname "$1")
+	local dst="$2/$dn"
+	mkdir -p "$dst"
+	mv "$1" "$dst/$bn"
+}
+
+_install() {
+	local bn=$(basename "$1")
+	local dn=$(dirname "$1")
+	for item in `find $dn -maxdepth 1 -name "$bn"`; do
```

Comment:
This is not a particularly safe construction. Consider using something like this:

```
find "$dn" -maxdepth 1 -name "$bn" | while IFS= read item; do _move_item "$item" "$2"; done
```

Reply:
I'll fix that, but can you share why this isn't safe?

Reply:
I can change that, no problem. However I would be grateful for explanation why `for` approach is unsafe?

Reply:
The path in `$dn` is not under your control and neither is the `find` output.

The easiest problem to understand with that `for` construction is word splitting. You're depending on splitting on `\n` for each line `find` prints, but word splitting also happens on any embedded spaces too.